### PR TITLE
feat(caregoal): add care goals to patient

### LINKED ISSUE #2080: Patient Care Goal
## 🚀 Feature Proposal
Add the ability to define care goals for a patient.

Care Goal is WHAT e.g. Decreases blood pressure by 15 points
Care Plan is HOW e.g. Exercise for 1 hour each day and decrease soda intake to 1 serving per week.

A care goal is defined as: 
- an identifier
- status: `proposed`, `planned`, `accepted`, `active`, `on hold`, `completed`, `cancelled`, `rejected`
- achievementStatus:`in progress`, `improving`, `worsening`, `no change`, `achieved`, `not achieved`, `no progress`, `not attainable`
- priority: `high`, `medium`, `low`
- description
- patient
- start date
- due date
- created date
- note

## Motivation
Define care goals for a patient

### PR DESCRIPTION
**Pull request by rsousaj**
Fixes #2080 .

**Changes proposed in this pull request:**

- Add ability to define care goals for a patient

### CODE REVIEW

File: src/__tests__/patients/care-goals/CareGoalForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    const noteInput = wrapper.findWhere((w) => w.prop('name') === 'note')
+
+    expect(alert).toHaveLength(1)
+    expect(alert.prop('message')).toEqual(expectedError.message)
+
+    expect(descriptionInput.prop('isInvalid')).toBeTruthy()
+    expect(descriptionInput.prop('feedback')).toEqual(expectedError.description)
+
+    expect(prioritySelector.prop('isInvalid')).toBeTruthy()
+    // expect(prioritySelector.prop('feedback')).toEqual(expectedError.priority)
```
Comment (jackcmeyer) [inline src/__tests__/patients/care-goals/CareGoalForm.test.tsx:275]: I think this commented out code should be tested. We should un-comment it and make sure the tests pass.

File: src/__tests__/patients/care-goals/ViewCareGoal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          <Route path="/patients/:id/care-goals/:careGoalId">
+            <ViewCareGoal />
+          </Route>
+        </Router>,
+      )
+    })
+
+    wrapper.update()
+
+    return { wrapper }
```
Comment (jackcmeyer) [inline src/__tests__/patients/care-goals/ViewCareGoal.test.tsx:38]: Change to `return { wrapper: wrapper as ReactWrapper }` so we can get type support

File: src/patients/hooks/useAddCareGoal.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import CareGoal from '../../shared/model/CareGoal'
+import { uuid } from '../../shared/util/uuid'
+import validateCareGoal from '../util/validate-caregoal'
+
+interface AddCareGoalRequest {
+  patientId: string
+  careGoal: Omit<CareGoal, 'id' | 'createdOn'>
+}
+
+async function addCareGoal(request: AddCareGoalRequest): Promise<CareGoal[]> {
```
Comment (jackcmeyer) [inline src/patients/hooks/useAddCareGoal.tsx:14]: there should be tests for this hook

**Comment by morrme** [comment]
> typing in the note field does not seem to work

confirmed--doesn't work for me either

File: src/__tests__/patients/care-goals/CareGoalForm.test.tsx (hunk 1 above)
Comment (rsousaj) [reply src/__tests__/patients/care-goals/CareGoalForm.test.tsx:275]: I think there is no feedback prop on SelectWithLabelFormGroup component. Also has a coment "_todo: add feedback in next round_"

File: src/patients/hooks/useAddCareGoal.tsx (hunk 3 above)
Comment (rsousaj) [reply src/patients/hooks/useAddCareGoal.tsx:14]: Yeah, I'd forgotten and then I did it on the lastest commit.

File: src/__tests__/patients/care-goals/ViewCareGoal.test.tsx (hunk 2 above)
Comment (rsousaj) [reply src/__tests__/patients/care-goals/ViewCareGoal.test.tsx:38]: Ok

File: src/__tests__/patients/view/ViewPatient.test.tsx [hunk 4]
```diff
@@ -131,7 +131,7 @@ describe('ViewPatient', () => {
     const tabs = tabsHeader.find(Tab)
     expect(tabsHeader).toHaveLength(1)
 
-    expect(tabs).toHaveLength(9)
+    expect(tabs).toHaveLength(10)
```
[the remark this answers is not in the document]
Comment (rsousaj) [reply src/__tests__/patients/view/ViewPatient.test.tsx:134]: Ok, tks

**Comment by rsousaj** [comment]
Tks guys. I'll make those changes.

**Review by fox1t — APPROVED** [review]
LGTM.
@jackcmeyer and @morrme are all comments addressed now?

**Review by jackcmeyer — APPROVED** [review]

**Pull request merged**